Mathlib/Data/Rat/NNRat.lean

```diff
@@ -208,12 +208,13 @@ def coeHom : ℚ≥0 →+* ℚ where
 
 @[simp, norm_cast]
 theorem coe_natCast (n : ℕ) : (↑(↑n : ℚ≥0) : ℚ) = n :=
-  map_natCast coeHom n
+  rfl
 #align nnrat.coe_nat_cast NNRat.coe_natCast
 
+-- See note [no_index around OfNat.ofNat]
 @[simp]
-theorem mk_coe_nat (n : ℕ) : @Eq ℚ≥0 (⟨(n : ℚ), n.cast_nonneg⟩ : ℚ≥0) n :=
```

Comment:
I think this might have been a hack to ensure the implicit argument to `Eq` was really `NNRat` and not `Subtype _`

Reply:
I would be surprised if it could implicitly turn into `Subtype _` because `NNRat` is not `reducible`. Was this more of a risk in Lean 3?

Reply:
It turns into `Subtype` because `⟨_, _⟩` is notation for `Subtype.mk` which has the wrong return type.

Reply:
Ah so this is the elusive difference between `(x : α)` and `@id α x`. Unless the former triggers a coercion, it just asserts that the type of `x` is defeq to `α`, whereas the latter actually transforms the type of the term. Is this defeq abuse a sign that this is a bad lemma and there should be `mk` function in the API of `ℚ≥0`? And I suppose using `@id ℚ≥0` or `show ℚ≥0 from` would both be inappropriate here because this is the rhs of a simp lemma so needs to be fully simplified..

Reply:
I think this lemma is reasonable, but it does suggest a flaw in the API (which is out of scope for this PR). If you don't like the `@Eq` approach, I think something like `(by exact ⟨_,_⟩) = _` will also assign the type correctly